src/topology/extremally_disconnected.lean

```diff
@@ -0,0 +1,124 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+import topology.compact_open
+import topology.stone_cech
+import topology.category.CompHaus
+
+/-!
+
+# Extremally disconnected spaces
+
+An extremally disconnected topological space is a space
+in which the closure of every open set is open.
+Such spaces are also called Stonean spaces.
+They are the projective objects in the category of compact Hausdorff spaces.
+
+Lemma `extremally_disconnected_of_projective` shows that compact Hausdorff spaces that are
+projective are extremally disconnected.  The converse is missing.
+
+# TODO
+
+Prove that a compact, Hausdorff, extremally disconnected space is projective.
```

Comment:
Maybe it's better to say "a projective object in the category of..." (what? Topological spaces? Compact Hausdorff spaces? etc)

Reply:
Kevin, thanks!  I expanded all the documentation for this file and clarified the point you raised.

In doing so, I am now slightly doubtful about the naming scheme.  The definition `compact_t2.projective` does not actually use the categorical definition of `projective`, i.e. `category_theory.projective` does not appear.  The lemma merely constructs the correct diagram and asserts the existence of a lift.  Should the name reflect the diagrammatic assumptions, or is the current name good?  In case my doubt is unclear, an example is the naming convention for lemmas such as `(a+b) ^ 2 = ...`: it is called `add_sq` instead of `binomial_sq` and would probably still be called this if `binomial` were defined, right?